### ISSUE TITLE
Probability threshold rather than 0.5

### ISSUE THREAD

**Issue by dopc** [Raschka-research-group/coral_pytorch#20]
Hey, congrats for this great project and thanks for sharing it!

## Question

What do you think about using a probability threshold rather than 0.5 in`corn_label_from_logits` function?

```
def corn_label_from_logits(logits, threshold=0.5):
    probas = torch.sigmoid(logits)
    probas = torch.cumprod(probas, dim=1)
    predict_levels = probas > threshold
    predicted_labels = torch.sum(predict_levels, dim=1)
    return predicted_labels
```

It could be something like that.

## Motivation
In some cases, tweaking this threshold might lead us to better classification objection result. However, I am not sure about if this is a proper change to make.

Best.

edit: grammar

**Comment by rasbt** [comment]
Sure, a threshold option could be added to both the CORAL and then CORN loss functions. In my opinion, I don't think they can improve the MAE but why not having this option. It's similar to sliding the threshold in binary classification to adjust the recall and precision.

**Comment by dopc** [comment]
Thanks for quick reply. 

---

> It's similar to sliding the threshold in binary classification to adjust the recall and precision.

Exactly!

---

> CORAL and then CORN loss functions

Which functions do you mean with these statements? `coral_loss()` and `corn_loss()` or `proba_to_label()` and `corn_label_from_logits`?

**Comment by rasbt** [comment]
Good point. I just said "loss functions" in a general sense, but looking at the code, that would be the `proba_to_label` and `corn_label_from_logits` functions

**Comment by dopc** [comment]
Got it. Thanks so much!
